src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -285,28 +324,38 @@ private static X509ChainStatusFlags MapVerifyErrorToChainStatus(Interop.libcrypt
             }
         }
 
-        internal static X509Certificate2Collection FindCandidates(
+        internal static List<X509Certificate2> FindCandidates(
             X509Certificate2 leaf,
-            X509Certificate2Collection extraStore)
+            X509Certificate2Collection extraStore,
+            List<X509Certificate2> downloaded,
+            ref TimeSpan remainingDownloadTime)
         {
-            X509Certificate2Collection candidates = new X509Certificate2Collection();
+            List<X509Certificate2> candidates = new List<X509Certificate2>();
             Queue<X509Certificate2> toProcess = new Queue<X509Certificate2>();
             toProcess.Enqueue(leaf);
 
-            using (var rootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
-            using (var intermediateStore = new X509Store(StoreName.CertificateAuthority, StoreLocation.LocalMachine))
+            using (var systemRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
```

Comment:
If it was capable of growing much more than this I'd probably agree; but I like having the explicit usings; and then all of their `Certificates` collections are already packed into an array and handled collectively past this point.